### PR TITLE
add endpoints field in ext discover api

### PR DESCRIPTION
**Pull request by shimish2**
distribution spec extension discover api has endpoints field required.

https://github.com/opencontainers/distribution-spec/blob/main/extensions/_oci.md#extensions-property-descriptions

Signed-off-by: Shivam Mishra <shimish2@cisco.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/api/controller_test.go [hunk 1]
```diff
@@ -4854,6 +4854,10 @@ func TestDistSpecExtensions(t *testing.T) {
 		err = json.Unmarshal(resp.Body(), &extensionList)
 		So(err, ShouldBeNil)
 		So(len(extensionList.Extensions), ShouldEqual, 1)
+		So(len(extensionList.Extensions[0].Endpoints), ShouldEqual, 1)
```
Comment (rchincha) [inline pkg/api/controller_test.go:4857]: Add the test for the actual endpoint match here?

File: pkg/api/controller_test.go [hunk 2]
```diff
@@ -4854,6 +4854,11 @@ func TestDistSpecExtensions(t *testing.T) {
...
+		So(extensionList.Extensions[0].Name, ShouldEqual, "search")
+		So(extensionList.Extensions[0].URL, ShouldContainSubstring, "_search.md")
+		So(extensionList.Extensions[0].Description, ShouldNotBeEmpty)
+		So(extensionList.Extensions[0].Endpoints[0], ShouldEqual, "_search")
```
Comment (rchincha) [inline pkg/api/controller_test.go:4861]: Not "/_search"? or "/v2/_search"?

File: pkg/api/controller_test.go (hunk 1 above)
Comment (shimish2) [reply pkg/api/controller_test.go:4857]: Yes, I will add.

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**